planner/core/task.go

```diff
@@ -1520,6 +1520,16 @@ type AggInfo struct {
 	Schema       *expression.Schema
 }
 
+// ExpandRetTypeForAvg expands the Flen of decimal type to avoid overflow
```

Comment:
I think it is a little hack, why not just infer the schema for `sum` function? This will make sure that the rewriten `sum` share the same schema with original `sum`.

Reply:
that should make the scope for `typeInfer()` to `TypeInfer()`